src/core/meshcop/joiner.cpp

```diff
@@ -86,6 +86,30 @@ void Joiner::SetState(otJoinerState aState)
     return;
 }
 
+otError Joiner::ValidatePskd(const char *aPskd)
+{
+    otError error      = OT_ERROR_INVALID_ARGS;
+    size_t  pskdLength = strlen(aPskd);
```

Comment:
use strnlen?

Reply:
1. we use `strlen` more in our code (by searching `strlen` vs `strnlen`).
2. `strnlen` hides the true length of `aPSKd`.
3. `strnlen` is not the C++ standard before C++11 (`std::strnLen_s`).

Reply:
Can use `ot::StringLength()` which is our internal `strnlen` 

We use it here in `commissioner` code for checking PSKd: https://github.com/openthread/openthread/blob/master/src/core/meshcop/commissioner.cpp#L288
(BTW, can we use the helper `Joiner::ValidatePskd()` in the above code as well?)

Reply:
@abtink 

> Can use ot::StringLength() which is our internal strnlen

Will do.

>BTW, can we use the helper Joiner::ValidatePskd() in the above code as well?

Yes, we did this. https://github.com/openthread/openthread/pull/4934/files#diff-adad1a583f73a28505dd375d3aaadd93